model/LtiListenerService.php

```diff
@@ -125,6 +126,27 @@ public function executionStateChanged(DeliveryExecutionState $event)
     }
 
     /**
+     * @param DeliveryRunEvent $event
+     */
+    public function deliveryRun(DeliveryRunEvent $event)
+    {
+        $delivery = $event->getDelivery();
+        $user = $event->getUser();
+        $remoteLink = \taoLti_models_classes_LtiService::singleton()->getLtiSession()->getLtiLinkResource();
+        $executions = LTIDeliveryTool::singleton()->getLinkedDeliveryExecutions($delivery, $remoteLink, $user->getIdentifier());
+        if (!empty($executions)) {
+            $deliveryExecutionService = $this->getServiceManager()->get(LtiDeliveryExecutionService::SERVICE_ID);
+            foreach ($executions as $deliveryExecution) {
+                if (!$deliveryExecutionService->isFinished($deliveryExecution) && $deliveryExecution->getState()->getUri() != DeliveryExecution::STATE_PAUSED) {
+                    /** @var DeliveryExecutionStateService $deliveryExecutionStateService */
+                    $deliveryExecutionStateService = $this->getServiceManager()->get(DeliveryExecutionStateService::SERVICE_ID);
+                    $deliveryExecutionStateService->pauseExecution($deliveryExecution);
```

Comment:
Also please consider to pass pause reason as the second parameter of `pauseExecution()`